turbopack/crates/turbopack-core/src/resolve/parse.rs

```diff
@@ -122,6 +126,32 @@ impl Request {
         })
     }
 
+    /// Turns the request into a pattern, similar to [Request::request()] but
+    /// more complete.
+    pub fn request_pattern(&self) -> Option<Pattern> {
```

Comment:
```suggestion
    pub fn request_pattern(&self) -> Pattern {
```

You can return a Pattern for Request::Alternatives too, so this method always returns a Pattern

Reply:
Good point, I had to make it `pub fn request_pattern(self: Vc<Self>) -> Result<Vc<Pattern>>` for that